aten/src/ATen/xpu/XPUEvent.h

```diff
@@ -0,0 +1,136 @@
+#pragma once
+#include <ATen/xpu/XPUContext.h>
+#include <c10/util/Optional.h>
+
+namespace at::xpu {
+
+/*
+ * XPUEvent are movable not copyable wrappers around SYCL event. XPUEvent are
+ * constructed lazily when first recorded. It has a device, and this device is
+ * acquired from the first recording stream. Later streams that record the event
+ * must match the same device.
+ */
+struct TORCH_XPU_API XPUEvent {
+  // Constructors
+  XPUEvent() noexcept = default;
+  XPUEvent(bool enable_timing) noexcept : enable_timing_{enable_timing} {}
+
+  ~XPUEvent() {}
+
+  XPUEvent(const XPUEvent&) = delete;
+  XPUEvent& operator=(const XPUEvent&) = delete;
+
+  XPUEvent(XPUEvent&& other) noexcept {
+    moveHelper(std::move(other));
+  }
+  XPUEvent& operator=(XPUEvent&& other) noexcept {
+    if (this != &other) {
+      moveHelper(std::move(other));
+    }
+    return *this;
+  }
+
+  operator sycl::event&() const {
+    return event();
+  }
+
+  optional<at::Device> device() const {
+    if (isCreated()) {
+      return at::Device(at::kXPU, device_index_);
+    } else {
+      return {};
+    }
+  }
+
+  bool isCreated() const {
+    return (event_.get() != nullptr);
+  }
+  DeviceIndex device_index() const {
+    return device_index_;
+  }
+  sycl::event& event() const {
+    return *event_;
+  }
+
+  bool query() const {
+    using namespace sycl::info;
+    if (!isCreated()) {
+      return true;
+    }
+
+    return event().get_info<event::command_execution_status>() ==
```

Comment:
Is this on par with cuda event, cudaEventQuery?

Reply:
yes, align to `cudaEventQuery`.